File: Recursion in C/palindrome.c
#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>
#include <string.h>
bool palindrome(char *input, int i, int j)
{
    // base case
    if (i == j - 1)
    {
        return true;
    }
    else if (input[i] == input[j - 1])
    {
        i++;
        j--;
        return palindrome(input, i, j);
    }
    else
    {
        return false;
    }
}
int main(void)
{
    char input[] = "racecar";
    int i = 0;
    int j = strlen(input);
    printf("%s", palindrome(input, i, j) ? "True" : "False");
}

File: Recursion in Js/palindrome.js
function palindrome(input, i, j) {
    if (i == j - 1) {
        return true;
    } else if (input[i] == input[j - 1]) {
        i++;
        j--;
        return palindrome(input, i, j);
    }
    else {
        return false;
    }
}

var input = "racecar";
var i = 0;
var j = input.length;
console.log(palindrome(input, i, j));